common/src/main/scala/uk/ac/wellcome/platform/stacks/common/services/CatalogueService.scala

```diff
@@ -1,158 +1,126 @@
 package uk.ac.wellcome.platform.stacks.common.services
 
-import cats.instances.future._
-import cats.instances.list._
-import cats.syntax.traverse._
-import com.google.gson.internal.LinkedTreeMap
-import uk.ac.wellcome.platform.catalogue
-import uk.ac.wellcome.platform.catalogue.models.{ItemIdentifiers, ResultListItems}
-import uk.ac.wellcome.platform.stacks.common.models._
-
-import scala.collection.JavaConverters._
-import scala.concurrent.{ExecutionContext, Future}
-import scala.util.Try
-
-class CatalogueService(baseUrl: Option[String])(
-  implicit
-  ec: ExecutionContext
-) {
-  protected val apiClient = new catalogue.ApiClient()
-
-  baseUrl.foreach { apiClient.setBasePath }
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.stream.ActorMaterializer
+import uk.ac.wellcome.platform.stacks.common.models.{StacksItem, _}
 
-  val worksApi = new catalogue.api.WorksApi(apiClient)
+import scala.concurrent.Future
 
-  type LTM[T] = LinkedTreeMap[String, T]
-
-  implicit class LinkedTreeMapExtractor(ltm: LTM[Any]) {
-    def safeGet[T](key: String): Option[T] =
-      Option(ltm.get(key)).flatMap(o => Try(o.asInstanceOf[T]).toOption)
+class CatalogueService(val baseUri: Uri = Uri(
+  "https://api.wellcomecollection.org/catalogue/v2"
+))(
+  implicit
+    val system: ActorSystem,
+    val mat: ActorMaterializer
+) extends AkkaClientService
+    with AkkaClientServiceGet {
+
+  import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+  import io.circe.generic.auto._
+  import CatalogueService._
+
+  protected def getIdentifier(
+                               identifiers: List[IdentifiersStub]
+                             ): Option[SierraItemIdentifier] = identifiers filter (
+    _.identifierType.id == "sierra-identifier"
+    ) match {
+    case List(IdentifiersStub(_, value)) =>
+      //TODO: This can fail!
```

Comment:
How can it fail? Can we provide a bit more explanation here?

Reply:
It's a conversion from `String` to `Long`, `.toLong` will throw an exception if you hand it nonesense.

Reply:
Ah, of course. So we're expecting numeric IDs from Sierra here?